component/s3/file/inout.go

```diff
@@ -0,0 +1,46 @@
+package s3
+
+import (
+	"time"
+
+	"github.com/grafana/agent/pkg/flow/rivertypes"
+)
+
+// Arguments implements the input for the s3 component
+type Arguments struct {
+	Path string `river:"path,attr"`
+	// PollFrequency determines the frequency to check for changes
+	// defaults to 10m
+	PollFrequency time.Duration `river:"poll_frequency,attr,optional"`
+	// IsSecret determines if the content should be displayed to the user
+	IsSecret bool `river:"is_secret,attr,optional"`
+	// Options allows you to override default settings
+	Options AWSOptions `river:"options,block,optional"`
```

Comment:
Having these settings in an options block feels a little strange. Aren't arguments already the options for a component? Should this just be a flat struct instead?

Reply:
How would you feel about naming them `aws_options`? They are overriding default behavior, so I don't want them flattened out.

Reply:
`client` or `client_options`, maybe? I'm not sure about the `aws_` prefix because there's a lot of non-AWS S3 implementations nowadays.

Reply:
I was thinking any non-aws would have it's own block ie `gcp_options` for custom overrides. TBF I dont know how many of those there actually are or even if we want them in the same component.

Reply:
I'd take a look at Loki's docs to see if its s3 storage backend needs any options that are specific to a vendor (I'd be surprised, because vendor-specific settings would probably not work well in the s3 community)

Reply:
They use s3, gcs, and azure object storage as separate blocks entirely with their specific settings. Granted they are likely using the specific providers instead of s3 compatible mode. https://grafana.com/docs/loki/latest/configuration/#storage